In [2]:
import numpy as np
import pandas as pd

In [6]:
mm_sales = pd.read_csv("Data/MegaMart_sales.csv")
mm_sales.head()

Order ID                                Product Name  Discount  \
0  AZ-2011-1029887  Novimex Color Coded Labels, 5000 Label Set       0.0   
1   AZ-2011-107716              Deflect-O Door Stop, Erganomic       0.0   
2  AZ-2011-1087704               Belkin Flash Drive, Bluetooth       0.0   
3  AZ-2011-1372644                  Panasonic Printer, Durable       0.0   
4  AZ-2011-1362199                   Sanford Pens, Fluorescent       0.5   

   Sales  Profit  Quantity         Category Sub-Category  
0     26       7         2  Office Supplies       Labels  
1     85      15         2        Furniture  Furnishings  
2    294     109         7       Technology  Accessories  
3    800     168         3       Technology     Machines  
4     25     -11         4  Office Supplies          Art

In [9]:
mm_new_sales =pd.read_csv("Data/MegaMart_newsales.csv")
mm_new_sales.head()

frames = [mm_sales, mm_new_sales]

main_frame = pd.concat(frames,axis=0,ignore_index=True)
main_frame.groupby(by='Category').Sales.sum()

Category
Furniture          2460
Office Supplies    7970
Technology         8588
Name: Sales, dtype: int64

In [10]:
highest_order=main_frame.loc[(main_frame['Sales']>250) & (main_frame['Profit']>50)]
highest_order=highest_order.sort_values(by='Sales',ascending=False) 
print(highest_order)

           Order ID                         Product Name  Discount  Sales  \
30  AZ-2011-1410648         Nokia Smart Phone, Full Size       0.0   1908   
58   AZ-2011-176674                Hoover Microwave, Red       0.1   1667   
8   AZ-2011-1174243     Nokia Audio Dock, with Caller ID       0.0   1334   
20  AZ-2011-1322840     Motorola Headset, with Caller ID       0.0    957   
28  AZ-2011-1372644           Panasonic Printer, Durable       0.0    800   
3   AZ-2011-1372644           Panasonic Printer, Durable       0.0    800   
18  AZ-2011-1406494         Fellowes Lockers, Industrial       0.1    748   
29  AZ-2011-1406494         Fellowes Lockers, Industrial       0.1    748   
39  AZ-2011-1536006      Logitech Keyboard, Programmable       0.0    666   
33  AZ-2011-1445262          Apple Smart Phone, Cordless       0.0    636   
31   AZ-2011-144325        Bush Stackable Bookrack, Pine       0.0    630   
14  AZ-2011-1260928        Eldon File Cart, Single Width       0.1    576   

In [11]:
lowest_profit=main_frame[main_frame.Profit>0]
lowest_profit=lowest_profit.loc[lowest_profit.Category=='Technology']
lowest_profit['Profit_per_qty']=lowest_profit['Profit']/lowest_profit['Quantity']
lowest_profit.sort_values(by='Profit_per_qty',ascending=True)

Order ID                      Product Name  Discount  Sales  \
39  AZ-2011-1536006   Logitech Keyboard, Programmable       0.0    666   
2   AZ-2011-1087704     Belkin Flash Drive, Bluetooth       0.0    294   
13  AZ-2011-1087704     Belkin Flash Drive, Bluetooth       0.0    294   
63  AZ-2011-1967754      Logitech Numeric Keypad, USB       0.0     93   
8   AZ-2011-1174243  Nokia Audio Dock, with Caller ID       0.0   1334   
20  AZ-2011-1322840  Motorola Headset, with Caller ID       0.0    957   
3   AZ-2011-1372644        Panasonic Printer, Durable       0.0    800   
28  AZ-2011-1372644        Panasonic Printer, Durable       0.0    800   
52  AZ-2011-1722024            Cisco Audio Dock, VoIP       0.0    364   
33  AZ-2011-1445262       Apple Smart Phone, Cordless       0.0    636   
30  AZ-2011-1410648      Nokia Smart Phone, Full Size       0.0   1908   

    Profit  Quantity    Category Sub-Category  Profit_per_qty  
39      66         9  Technology  Accessories        7.333333  
2      109         7  Technology  Accessories       15.571429  
13     109         7  Technology  Accessories       15.571429  
63      40         2  Technology  Accessories       20.000000  
8      200         8  Technology       Phones       25.000000  
20     316        12  Technology       Phones       26.333333  
3      168         3  Technology     Machines       56.000000  
28     168         3  Technology     Machines       56.000000  
52     142         2  Technology       Phones       71.000000  
33     140         1  Technology       Phones      140.000000  
30     820         3  Technology       Phones      273.333333

#### Combining two data frames
Question 1: Some of the orders are stored in another csv file named `megamart_new sales`. Read the csv file, store it in a data frame and add it to the `megamart_sales` data frame. Find the total sales value of the category 'Office Supplies'  after combining the dataframes

    a)7970
    b)6964
    c)7494
    d)6076

In [ ]:
#your code here


#### Dropping duplicates
Question 2: There are some duplicate rows in the data frame. Drop these rows and calculate the total sales value of the category Office Supplies.

    a)7156
    b)6496
    c)6964
    d)6023

In [ ]:
#your code here



#### Best category-sub category 
Question 3: Find the most profitable category and sub category combination based on the net profit.


    a)Furniture-Bookcases
    b)Office supplies-Appliances
    c)Office supplies-Storage
    d)Technology-Phones

In [ ]:
#your code here



#### Invalid order IDs
Question 4: How many invalid order IDs are there in the data frame. An order id is of the form AZ-2011-Y where Y represents a whole number. A Order ID is said to be valid only if Y consists of 7 digits. Find the number of valid order order IDs in the data frame.
    
    a)6
    b)7
    c)8
    d)9


In [ ]:
#your code here



#### Occurence of furniture in top 25 sales
Question 5: Find the top 25 orders based on sales value and find the number of orders which belong to furniture category.

    a)2
    b)3
    c)4
    d)5 

In [ ]:
#your code here



#### And operation
Question 6: Among the orders with sales>250 and profit>50, find the product name of the fourth highest order based on sales value.

    a)Motorola Headset, with Caller ID
    b)Panasonic Printer, Durable	
    c)Hoover Microwave, Red	
    d)Fellowes Lockers, Industrial	

In [ ]:
#your code here



#### Column manipulation
Question 7: Remove the orders with negative profit by dropping the corresponding rows with negative `Profit`. Find the product that makes the lowest profit per Quantity in the Technology category.

    a) Nokia Audio Dock, with Caller ID
    b) Logitech Keyboard, Programmable
    c) Motorola Headset, with Caller ID
    d) Belkin Flash Drive, Bluetooth

In [ ]:
#your code here


